README.md
fantomidemodule
===============

Netbeans plugin module to support the Fantom Language (Also used by standalone fantomide project)

See more info about the FantomIde here: http://fantomide.colar.net/

Note that I also maintain a much leaner pure Fantom IDE called Camembert: http://www.status302.com/camembert


doc/history.txt
Introducing NB Api's / CSL

Introducing Fantom

Initial: nb_fab (plugin)  r575
- create plugin prj
- layer.xml
- mime.xml - schlieman ! .nbs

Revision <592/593 : Fan_antlr
- Fan.g
- nbproject -> working project
- build task (build.xml) -> FanLexer / FanParser 

Rev <595:
- antlr-runtime lib

Rev < 602: more grammar work

Rev <603: 
- antlr.sh - testing grammar
- working with antlrworks -> testing/debugging grammar
- antlr works : extra memory

Rev <607:
- Working grammar
- test fan files

Rev <623:
- More grammar work - line breaks hack, maps and other tricky structures

Rev <636:
- More grammar work - line breaks hack, 

Rev <694:
- More grammar edge cases work
- Working with small grammar subsets to find issues Test.g / antlrworks

Rev 700:
- updated ant build file

Rev 704:
- created fan suite

Rev 705:
- Proper NB plugin
- Integrated grammar into Fan plugin
- Moved Fan_antlr into Fan /  cleanup

Rev < 712:
- FanLangHierarchy / FanLanguage.java / AntlrTokensToNetbeans.java / NBFanLexer.java


Rev 712:
- textcolors.xml
- layer.xml
- fanresolver.xml

Rev 719:
- CSL
- Lexer integration

Rev 724:
- error recovery

Rev 733:
- Parser integration / crude error reporting from antlr

Rev 736:
- Module distro (plugin) added to FanSuite (nbm)

Rev 739:
- First release as an NBM !

Rev < 744:
- More error recovery work

Rev < 746:
- started ast work
- navigator window

Rev 752: 
- working navigator window : got to class/method etc...

Rev 754:
- Code folding

Rev 757:
- Fan project support: logical view, filter node

Rev < 785:
- Project creation wizzrad / UI
- Project Execution (run)

Rev < 786:
- project execution
- fanshell support

Rev < 815:
TODO: explain gridlayout java 1.6 issue

- settings window
- Manu acions (build, run etc..)
- custom icons for files etc...

Rev < 819:
- Create a release (suite)
- added branding / images /etc...

Rev < 820:
- using jot for wizard templates (freemarker issues)

832: Open automatically new files in editor

839: dependencies cleanup (explain), release suite

847: plugin folder

852: Licence , keystroke handler

855: indentation
<866: bracket matching / autoclosing items

869: pretty formatter

882: Highlighting in interpolated strings

889/890: fixes to navigator glitches, Fix to files dealyed appareance in project view

901: Initial support for build, run and build&run of pods

904: more support for pods: build etc...

911: self signed certificate for plugin (plugin portal)

912: releases , osx release

924: java 1.5 wizzard issue

928: started completion work

931: license work

936: started debugger

945-950: breakpoints

956: branding: splash, icon, version, license

976: working debugger

<1015: code formatter fixes

1000+: completion work: lucene / no etc...
new ast / parser - scoping work etc...
custom db backed index
Field parsing, slots & type determination
using types resolving
Simplified resolving types
fandoc for slot proposals
complete call expressions
expression completion
propose variables
propose correct types and static or dynamic slots according to context
local vars / scoping

1162: wrapped jars
perm gen space

indexer work

1224:
Added customized CslJar.java so I can build cslJar from binaries and not need NB sources.

<1235: java completion

<1242: type resolution work

<1250: structureanalyzewr using scope rewrite

1262: type inference

-----------------------------------------------------------------
date:        Fri Dec 17 09:29:32 2010 -0800
summary:     Started work on (new) tales integration

date:        Thu Dec 16 15:55:59 2010 -0800
summary:     Working method/field type creation Hints:

date:        Fri Dec 10 11:37:09 2010 -0800
summary:     Hint/fixes providers for:

date:        Thu Dec 09 15:40:32 2010 -0800
summary:     Improved completion of slots (propose params, expand closures)

date:        Thu Dec 09 15:38:42 2010 -0800
summary:     Added enumeration to error reporting, to be able to pinpoint errors better and use it for hints.

date:        Wed Dec 08 10:20:23 2010 -0800
summary:     Fixed issue 14 (New file not appearing righ away in project/File navigators.)

date:        Tue Dec 07 12:40:32 2010 -0800
summary:     Completion fixes/enhancements - propose params/closures

date:        Mon Nov 01 16:20:40 2010 -0700
summary:     JVM settings (permspace)

date:        Tue Aug 24 14:55:55 2010 -0700
summary:     Updated so that it works with NB 6.9.1

date:        Fri May 28 09:05:12 2010 -0700
summary:     Debugger classpath fix

date:        Mon Jun 07 11:13:21 2010 -0700
summary:     Modified fileToSnaphot to circunvent odd classcastexception with nb 6.9 (bug ??)

date:        Fri Jun 04 16:44:21 2010 -0700
summary:     Completed the rewrite of the AST construction to prevent memory leak.

date:        Wed Jun 02 14:36:35 2010 -0700
summary:     Early work on trying to make parser task cancellable, which would really help performance.

date:        Wed May 12 15:22:12 2010 -0700
summary:     Normalizing files

date:        Thu Apr 29 10:09:44 2010 -0700
summary:     Require restart after install

date:        Wed Apr 28 16:46:09 2010 -0700
summary:     Started some work on the actions

date:        Wed Apr 28 10:17:51 2010 -0700
summary:     Registering the Option panel through new annotation instead of deprecated API

date:        Tue Apr 20 05:27:17 2010 +0000
summary:     protected from a NPE when file cannot be parsed at all.

date:        Tue Apr 20 04:53:39 2010 +0000
summary:     Fix to node action project finder (build/debug etc..) such as it's smarter at finding the right project to run the task against.

date:        Tue Apr 20 04:19:26 2010 +0000

src/net/colar/netbeans/fan/FanTokens.java
/*Automatically generated by AntlrTokensToNetbeans.java*/

package net.colar.netbeans.fan;

import java.util.Hashtable;

public class FanTokens{

private static Hashtable<Integer, FanTokenID> tokens=new Hashtable<Integer, FanTokenID>();

static{ // initializes tokens
    //Generated from ANTLR tokens: src/net/colar/netbeans/fan/antlr/Fan.tokens

    FanTokenID token;

	token=new FanTokenID("EXPONENT", 163, "");
	tokens.put(new Integer(163),token);
	token=new FanTokenID("AST_INDEX_EXPR", 81, "");
	tokens.put(new Integer(81),token);
	token=new FanTokenID("DSL", 124, "");
	tokens.put(new Integer(124),token);
	token=new FanTokenID("OP_AND", 109, "operator");
	tokens.put(new Integer(109),token);
	token=new FanTokenID("KW_NATIVE", 22, "keyword");
	tokens.put(new Integer(22),token);
	token=new FanTokenID("AST_CATCH_DEF", 77, "");
	tokens.put(new Integer(77),token);
	token=new FanTokenID("LETTER", 165, "");
	tokens.put(new Integer(165),token);
	token=new FanTokenID("HEXNB", 158, "");
	tokens.put(new Integer(158),token);
	token=new FanTokenID("CHAR", 128, "character");
	tokens.put(new Integer(128),token);
	token=new FanTokenID("OP_2PLUS", 122, "operator");
	tokens.put(new Integer(122),token);
	token=new FanTokenID("KW_DEFAULT", 14, "keyword");
	tokens.put(new Integer(14),token);
	token=new FanTokenID("KW_ONCE", 37, "keyword");
	tokens.put(new Integer(37),token);
	token=new FanTokenID("SP_SEMI", 104, "separator");
	tokens.put(new Integer(104),token);
	token=new FanTokenID("INC_UNKNOWN_ITEM", 166, "error");
	tokens.put(new Integer(166),token);
	token=new FanTokenID("AST_MIXIN", 53, "");
	tokens.put(new Integer(53),token);
	token=new FanTokenID("T__167", 167, "");
	tokens.put(new Integer(167),token);
	token=new FanTokenID("KW_PUBLIC", 34, "keyword");
	tokens.put(new Integer(34),token);
	token=new FanTokenID("KW_ISNOT", 42, "keyword");
	tokens.put(new Integer(42),token);
	token=new FanTokenID("OP_MOD", 119, "operator");
	tokens.put(new Integer(119),token);
	token=new FanTokenID("PAR_R", 138, "");
	tokens.put(new Integer(138),token);
	token=new FanTokenID("OP_MINUS", 116, "operator");
	tokens.put(new Integer(116),token);
	token=new FanTokenID("KW_NEW", 36, "keyword");
	tokens.put(new Integer(36),token);
	token=new FanTokenID("KW_BREAK", 4, "keyword");
	tokens.put(new Integer(4),token);
	token=new FanTokenID("KW_SWITCH", 9, "keyword");
	tokens.put(new Integer(9),token);
	token=new FanTokenID("AST_INC_DOTCALL", 83, "");
	tokens.put(new Integer(83),token);
	token=new FanTokenID("OP_RANG_EXCL_OLD", 112, "operator");
	tokens.put(new Integer(112),token);
	token=new FanTokenID("AST_CLASS", 51, "");
	tokens.put(new Integer(51),token);
	token=new FanTokenID("OP_BITOR", 152, "operator");
	tokens.put(new Integer(152),token);
	token=new FanTokenID("QUOTSTR", 129, "string");
	tokens.put(new Integer(129),token);
	token=new FanTokenID("AST_TYPE", 90, "");
	tokens.put(new Integer(90),token);
	token=new FanTokenID("OP_BANG", 121, "operator");
	tokens.put(new Integer(121),token);
	token=new FanTokenID("AST_DOT_CALL", 63, "");
	tokens.put(new Integer(63),token);
	token=new FanTokenID("AST_TYPE_LIT", 69, "");
	tokens.put(new Integer(69),token);
	token=new FanTokenID("PAR_L", 137, "");
	tokens.put(new Integer(137),token);
	token=new FanTokenID("OP_MULTI", 117, "operator");
	tokens.put(new Integer(117),token);
	token=new FanTokenID("AST_USING_POD", 66, "");
	tokens.put(new Integer(66),token);
	token=new FanTokenID("KEYWORD", 151, "");
	tokens.put(new Integer(151),token);
	token=new FanTokenID("KW_PROTECTED", 33, "keyword");
	tokens.put(new Integer(33),token);
	token=new FanTokenID("AST_STR", 60, "");
	tokens.put(new Integer(60),token);
	token=new FanTokenID("LINE_COMMENT", 141, "comment");
	tokens.put(new Integer(141),token);
	token=new FanTokenID("CP_COMPARATORS", 111, "comparator");
	tokens.put(new Integer(111),token);
	token=new FanTokenID("NUMBER", 134, "number");
	tokens.put(new Integer(134),token);
	token=new FanTokenID("AS_EQUAL", 106, "assignment");
	tokens.put(new Integer(106),token);
	token=new FanTokenID("KW_CATCH", 15, "keyword");
	tokens.put(new Integer(15),token);
	token=new FanTokenID("UNDERSCORE", 153, "");
	tokens.put(new Integer(153),token);
	token=new FanTokenID("OP_DIV", 118, "operator");
	tokens.put(new Integer(118),token);
	token=new FanTokenID("SP_COMMA", 98, "separator");
	tokens.put(new Integer(98),token);
	token=new FanTokenID("AST_SYMBOL", 71, "");
	tokens.put(new Integer(71),token);
	token=new FanTokenID("AST_LIST", 73, "");
	tokens.put(new Integer(73),token);
	token=new FanTokenID("URI", 127, "string");
	tokens.put(new Integer(127),token);
	token=new FanTokenID("KW_FINALLY", 16, "keyword");
	tokens.put(new Integer(16),token);
	token=new FanTokenID("WS", 140, "");
	tokens.put(new Integer(140),token);
	token=new FanTokenID("KW_THROW", 10, "keyword");
	tokens.put(new Integer(10),token);
	token=new FanTokenID("OP_POUND", 131, "operator");
	tokens.put(new Integer(131),token);
	token=new FanTokenID("KW_STATIC", 21, "keyword");
	tokens.put(new Integer(21),token);
	token=new FanTokenID("INC_COMMENT", 49, "error");
	tokens.put(new Integer(49),token);
	token=new FanTokenID("AST_METHOD", 54, "");
	tokens.put(new Integer(54),token);
	token=new FanTokenID("AST_ENUM", 52, "");
	tokens.put(new Integer(52),token);
	token=new FanTokenID("AST_CALL", 61, "");
	tokens.put(new Integer(61),token);
	token=new FanTokenID("AST_CTOR_BLOCK", 75, "");
	tokens.put(new Integer(75),token);
	token=new FanTokenID("SP_QMARK", 99, "separator");
	tokens.put(new Integer(99),token);
	token=new FanTokenID("AST_ID", 86, "");
	tokens.put(new Integer(86),token);
	token=new FanTokenID("HEXHEADER", 161, "");
	tokens.put(new Integer(161),token);
	token=new FanTokenID("AS_INIT_VAL", 96, "assignment");
	tokens.put(new Integer(96),token);
	token=new FanTokenID("COMPL_DSL", 145, "");
	tokens.put(new Integer(145),token);
	token=new FanTokenID("NBTYPE", 164, "");
	tokens.put(new Integer(164),token);
	token=new FanTokenID("KW_SUPER", 39, "keyword");
	tokens.put(new Integer(39),token);
	token=new FanTokenID("OP_2MINUS", 123, "operator");
	tokens.put(new Integer(123),token);
	token=new FanTokenID("AST_SCOPE", 85, "");
	tokens.put(new Integer(85),token);
	token=new FanTokenID("OP_RSHIFT", 155, "operator");
	tokens.put(new Integer(155),token);
	token=new FanTokenID("KW_ELSE", 40, "keyword");
	tokens.put(new Integer(40),token);
	token=new FanTokenID("AST_FUNC_TYPE", 80, "");
	tokens.put(new Integer(80),token);
	token=new FanTokenID("OP_OR", 108, "operator");
	tokens.put(new Integer(108),token);
	token=new FanTokenID("AST_MODIFIER", 87, "");
	tokens.put(new Integer(87),token);
	token=new FanTokenID("DOC", 133, "");
	tokens.put(new Integer(133),token);
	token=new FanTokenID("KW_OVERRIDE", 24, "keyword");
	tokens.put(new Integer(24),token);
	token=new FanTokenID("AST_FIELD", 56, "");
	tokens.put(new Integer(56),token);
	token=new FanTokenID("AST_CODE_BLOCK", 58, "");
	tokens.put(new Integer(58),token);
	token=new FanTokenID("KW_ABSTRACT", 31, "keyword");
	tokens.put(new Integer(31),token);
	token=new FanTokenID("MULTI_COMMENT", 144, "comment");
	tokens.put(new Integer(144),token);
	token=new FanTokenID("KW_CONST", 20, "keyword");
	tokens.put(new Integer(20),token);
	token=new FanTokenID("AST_DOCS", 59, "");
	tokens.put(new Integer(59),token);
	token=new FanTokenID("COMPL_ML_COMMENT", 143, "");
	tokens.put(new Integer(143),token);
	token=new FanTokenID("OP_ELVIS", 105, "operator");
	tokens.put(new Integer(105),token);
	token=new FanTokenID("BRACKET_R", 95, "");
	tokens.put(new Integer(95),token);
	token=new FanTokenID("KW_TRUE", 45, "keyword");
	tokens.put(new Integer(45),token);
	token=new FanTokenID("ID", 126, "");
	tokens.put(new Integer(126),token);
	token=new FanTokenID("SP_COLON", 97, "separator");
	tokens.put(new Integer(97),token);
	token=new FanTokenID("AST_CAST", 79, "");
	tokens.put(new Integer(79),token);
	token=new FanTokenID("AST_CONSTRUCTOR_CHAIN", 57, "");
	tokens.put(new Integer(57),token);
	token=new FanTokenID("AST_TERM_EXPR", 62, "");
	tokens.put(new Integer(62),token);
	token=new FanTokenID("BRACKET_L", 94, "");
	tokens.put(new Integer(94),token);
	token=new FanTokenID("AT", 132, "");
	tokens.put(new Integer(132),token);
	token=new FanTokenID("STR", 130, "string");
	tokens.put(new Integer(130),token);
	token=new FanTokenID("OP_RANGE_EXCL", 113, "operator");
	tokens.put(new Integer(113),token);
	token=new FanTokenID("SP_PIPE", 101, "separator");
	tokens.put(new Integer(101),token);
	token=new FanTokenID("OP_TILDA", 156, "operator");
	tokens.put(new Integer(156),token);
	token=new FanTokenID("COMPL_QSTR", 148, "");
	tokens.put(new Integer(148),token);
	token=new FanTokenID("KW_VIRTUAL", 25, "keyword");
	tokens.put(new Integer(25),token);
	token=new FanTokenID("CP_EQUALITY", 110, "comparator");
	tokens.put(new Integer(110),token);
	token=new FanTokenID("AST_SAFE_DOT_CALL", 64, "");
	tokens.put(new Integer(64),token);
	token=new FanTokenID("KW_FOR", 6, "keyword");
	tokens.put(new Integer(6),token);
	token=new FanTokenID("KW_WHILE", 11, "keyword");
	tokens.put(new Integer(11),token);
	token=new FanTokenID("COMPL_URI", 150, "");
	tokens.put(new Integer(150),token);
	token=new FanTokenID("AST_LOCAL_DEF", 68, "");
	tokens.put(new Integer(68),token);
	token=new FanTokenID("AST_SLOT_LIT", 70, "");
	tokens.put(new Integer(70),token);
	token=new FanTokenID("OP_RANGE", 114, "operator");
	tokens.put(new Integer(114),token);
	token=new FanTokenID("KW_RETURN", 8, "keyword");
	tokens.put(new Integer(8),token);
	token=new FanTokenID("INC_DSL", 50, "error");
	tokens.put(new Integer(50),token);
	token=new FanTokenID("SQ_BRACKET_R", 136, "");
	tokens.put(new Integer(136),token);
	token=new FanTokenID("AST_NAMED_SUPER", 72, "");
	tokens.put(new Integer(72),token);
	token=new FanTokenID("KW_IS", 41, "keyword");
	tokens.put(new Integer(41),token);
	token=new FanTokenID("DIGIT", 146, "");
	tokens.put(new Integer(146),token);
	token=new FanTokenID("KW_TRY", 12, "keyword");
	tokens.put(new Integer(12),token);
	token=new FanTokenID("KW_IT", 44, "keyword");
	tokens.put(new Integer(44),token);
	token=new FanTokenID("DOT", 93, "");
	tokens.put(new Integer(93),token);
	token=new FanTokenID("KW_THIS", 38, "keyword");
	tokens.put(new Integer(38),token);
	token=new FanTokenID("KW_IF", 7, "keyword");
	tokens.put(new Integer(7),token);
	token=new FanTokenID("OP_CURRY", 120, "operator");
	tokens.put(new Integer(120),token);
	token=new FanTokenID("SQ_BRACKET_L", 135, "");
	tokens.put(new Integer(135),token);
	token=new FanTokenID("AST_INC_SAFEDOTCALL", 84, "");
	tokens.put(new Integer(84),token);
	token=new FanTokenID("OP_PLUS", 115, "operator");
	tokens.put(new Integer(115),token);
	token=new FanTokenID("KW_FALSE", 46, "keyword");
	tokens.put(new Integer(46),token);
	token=new FanTokenID("KW_INTERNAL", 35, "keyword");
	tokens.put(new Integer(35),token);
	token=new FanTokenID("KW_FINAL", 26, "keyword");
	tokens.put(new Integer(26),token);
	token=new FanTokenID("HEXLETTER", 147, "");
	tokens.put(new Integer(147),token);
	token=new FanTokenID("KW_CONTINUE", 5, "keyword");
	tokens.put(new Integer(5),token);
	token=new FanTokenID("KW_PRIVATE", 32, "keyword");
	tokens.put(new Integer(32),token);
	token=new FanTokenID("KW_ENUM", 29, "keyword");
	tokens.put(new Integer(29),token);
	token=new FanTokenID("HEX_HEADER", 157, "");
	tokens.put(new Integer(157),token);
	token=new FanTokenID("OP_SAFEDYN_CALL", 102, "operator");
	tokens.put(new Integer(102),token);
	token=new FanTokenID("LIST_TYPE", 100, "");
	tokens.put(new Integer(100),token);
	token=new FanTokenID("KW_VOLATILE", 23, "keyword");
	tokens.put(new Integer(23),token);
	token=new FanTokenID("KW_CASE", 13, "keyword");
	tokens.put(new Integer(13),token);
	token=new FanTokenID("FRACTIONAL", 160, "");
	tokens.put(new Integer(160),token);
	token=new FanTokenID("OP_LSHIFT", 154, "operator");
	tokens.put(new Integer(154),token);
	token=new FanTokenID("EXEC_COMMENT", 142, "comment");
	tokens.put(new Integer(142),token);
	token=new FanTokenID("KW_MIXIN", 30, "keyword");
	tokens.put(new Integer(30),token);
	token=new FanTokenID("AST_IT_BLOCK", 74, "");
	tokens.put(new Integer(74),token);
	token=new FanTokenID("AST_CHILD", 91, "");
	tokens.put(new Integer(91),token);
	token=new FanTokenID("OP_ARROW", 103, "operator");
	tokens.put(new Integer(103),token);
	token=new FanTokenID("KW_VOID", 27, "keyword");
	tokens.put(new Integer(27),token);
	token=new FanTokenID("DECIMAL", 159, "");
	tokens.put(new Integer(159),token);
	token=new FanTokenID("INC_URI", 48, "error");
	tokens.put(new Integer(48),token);
	token=new FanTokenID("KW_CLASS", 28, "keyword");
	tokens.put(new Integer(28),token);
	token=new FanTokenID("AS_ASSIGN_OP", 107, "assignment");
	tokens.put(new Integer(107),token);
	token=new FanTokenID("AST_MAP", 67, "");
	tokens.put(new Integer(67),token);
	token=new FanTokenID("AST_CONSTRUCTOR", 55, "");
	tokens.put(new Integer(55),token);
	token=new FanTokenID("KW_RD_ONLY", 19, "keyword");
	tokens.put(new Integer(19),token);
	token=new FanTokenID("INC_STR", 47, "error");
	tokens.put(new Integer(47),token);
	token=new FanTokenID("AST_STATIC_CALL", 65, "");
	tokens.put(new Integer(65),token);
	token=new FanTokenID("OP_SAFE_CALL", 125, "operator");
	tokens.put(new Integer(125),token);
	token=new FanTokenID("KW_AS", 18, "keyword");
	tokens.put(new Integer(18),token);
	token=new FanTokenID("COMPL_STR", 149, "");
	tokens.put(new Integer(149),token);
	token=new FanTokenID("LB", 139, "");
	tokens.put(new Integer(139),token);
	token=new FanTokenID("AST_FORMAL", 78, "");
	tokens.put(new Integer(78),token);
	token=new FanTokenID("KW_USING", 17, "keyword");
	tokens.put(new Integer(17),token);
	token=new FanTokenID("KW_NULL", 43, "keyword");
	tokens.put(new Integer(43),token);
	token=new FanTokenID("AST_FOR_INIT", 76, "");
	tokens.put(new Integer(76),token);
	token=new FanTokenID("AST_PARAM", 89, "");
	tokens.put(new Integer(89),token);
	token=new FanTokenID("FRACTION", 162, "");
	tokens.put(new Integer(162),token);
	token=new FanTokenID("AST_INHERITANCE", 88, "");
	tokens.put(new Integer(88),token);
	token=new FanTokenID("AST_INC_USING", 82, "");
	tokens.put(new Integer(82),token);
	token=new FanTokenID("SP_COLCOL", 92, "separator");
	tokens.put(new Integer(92),token);
	token=tokens.get(new Integer(29));
	token.setFixedText("enum");
	tokens.put(new Integer(29), token);
	token=tokens.get(new Integer(16));
	token.setFixedText("finally");
	tokens.put(new Integer(16), token);
	token=tokens.get(new Integer(37));
	token.setFixedText("once");
	tokens.put(new Integer(37), token);
	token=tokens.get(new Integer(35));
	token.setFixedText("internal");
	tokens.put(new Integer(35), token);
	token=tokens.get(new Integer(28));
	token.setFixedText("class");
	tokens.put(new Integer(28), token);
	token=tokens.get(new Integer(36));
	token.setFixedText("new");
	tokens.put(new Integer(36), token);
	token=tokens.get(new Integer(9));
	token.setFixedText("switch");
	tokens.put(new Integer(9), token);
	token=tokens.get(new Integer(43));
	token.setFixedText("null");
	tokens.put(new Integer(43), token);
	token=tokens.get(new Integer(20));
	token.setFixedText("const");
	tokens.put(new Integer(20), token);
	token=tokens.get(new Integer(38));
	token.setFixedText("this");
	tokens.put(new Integer(38), token);
	token=tokens.get(new Integer(10));
	token.setFixedText("throw ");
	tokens.put(new Integer(10), token);
	token=tokens.get(new Integer(23));
	token.setFixedText("volatile");
	tokens.put(new Integer(23), token);
	token=tokens.get(new Integer(8));
	token.setFixedText("return");
	tokens.put(new Integer(8), token);
	token=tokens.get(new Integer(33));
	token.setFixedText("protected");
	tokens.put(new Integer(33), token);
	token=tokens.get(new Integer(6));
	token.setFixedText("for");
	tokens.put(new Integer(6), token);
	token=tokens.get(new Integer(42));
	token.setFixedText("isnot");
	tokens.put(new Integer(42), token);
	token=tokens.get(new Integer(25));
	token.setFixedText("virtual");
	tokens.put(new Integer(25), token);
	token=tokens.get(new Integer(41));
	token.setFixedText("is");
	tokens.put(new Integer(41), token);
	token=tokens.get(new Integer(100));
	token.setFixedText("[]");
	tokens.put(new Integer(100), token);
	token=tokens.get(new Integer(39));
	token.setFixedText("super");
	tokens.put(new Integer(39), token);
	token=tokens.get(new Integer(167));
	token.setFixedText("$");
	tokens.put(new Integer(167), token);
	token=tokens.get(new Integer(13));
	token.setFixedText("case");
	tokens.put(new Integer(13), token);
	token=tokens.get(new Integer(21));
	token.setFixedText("static");
	tokens.put(new Integer(21), token);
	token=tokens.get(new Integer(45));
	token.setFixedText("true");
	tokens.put(new Integer(45), token);
	token=tokens.get(new Integer(40));
	token.setFixedText("else");
	tokens.put(new Integer(40), token);
	token=tokens.get(new Integer(15));
	token.setFixedText("catch");
	tokens.put(new Integer(15), token);
	token=tokens.get(new Integer(31));
	token.setFixedText("abstract");
	tokens.put(new Integer(31), token);
	token=tokens.get(new Integer(26));
	token.setFixedText("final");
	tokens.put(new Integer(26), token);
	token=tokens.get(new Integer(24));
	token.setFixedText("override");
	tokens.put(new Integer(24), token);
	token=tokens.get(new Integer(22));
	token.setFixedText("native");
	tokens.put(new Integer(22), token);
	token=tokens.get(new Integer(4));
	token.setFixedText("break");
	tokens.put(new Integer(4), token);
	token=tokens.get(new Integer(19));
	token.setFixedText("readonly");
	tokens.put(new Integer(19), token);
	token=tokens.get(new Integer(18));
	token.setFixedText("as");
	tokens.put(new Integer(18), token);
	token=tokens.get(new Integer(27));
	token.setFixedText("void");
	tokens.put(new Integer(27), token);
	token=tokens.get(new Integer(12));
	token.setFixedText("try");
	tokens.put(new Integer(12), token);
	token=tokens.get(new Integer(32));
	token.setFixedText("private");
	tokens.put(new Integer(32), token);
	token=tokens.get(new Integer(46));
	token.setFixedText("false");
	tokens.put(new Integer(46), token);
	token=tokens.get(new Integer(30));
	token.setFixedText("mixin");
	tokens.put(new Integer(30), token);
	token=tokens.get(new Integer(5));
	token.setFixedText("continue");
	tokens.put(new Integer(5), token);
	token=tokens.get(new Integer(7));
	token.setFixedText("if");
	tokens.put(new Integer(7), token);
	token=tokens.get(new Integer(44));
	token.setFixedText("it");
	tokens.put(new Integer(44), token);
	token=tokens.get(new Integer(11));
	token.setFixedText("while");
	tokens.put(new Integer(11), token);
	token=tokens.get(new Integer(34));
	token.setFixedText("public");
	tokens.put(new Integer(34), token);
	token=tokens.get(new Integer(14));
	token.setFixedText("default");
	tokens.put(new Integer(14), token);
	token=tokens.get(new Integer(17));
	token.setFixedText("using");
	tokens.put(new Integer(17), token);
	}

	public static Hashtable<Integer, FanTokenID> getTokens() {return tokens; }

}
